core/src/main/java/org/apache/calcite/rel/rules/JoinToCorrelateRule.java

```diff
@@ -64,10 +63,10 @@
 public class JoinToCorrelateRule extends RelOptRule {
 
   /**
-   * Function to extract the {@link org.apache.calcite.sql.SemiJoinType} parameter
+   * Function to extract the {@link JoinRelType} parameter
    * for the creation of the {@link org.apache.calcite.rel.logical.LogicalCorrelate}
    */
-  private final Function<Join, SemiJoinType> semiJoinTypeExtractor;
+  private final Function<Join, JoinRelType> semiJoinTypeExtractor;
```

Comment:
I have the impression that after this big refactoring the `semiJoinTypeExtractor` is not needed and since it is private it can be removed.

Reply:
I agree, will remove `semiJoinTypeExtractor`.